test/integration/targets/strategy_host_pinned/callback_plugins/callback_host_count.py

```diff
@@ -0,0 +1,39 @@
+# (c) 2024 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import annotations
+
+from ansible.plugins.callback import CallbackBase
+
+
+class CallbackModule(CallbackBase):
+    CALLBACK_VERSION = 2.0
+    CALLBACK_TYPE = 'stdout'
+    CALLBACK_NAME = 'callback_host_count'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._executing_hosts_counter = 0
+
+    def v2_playbook_on_task_start(self, task, is_conditional):
+        self._display.display(task.name or task.action)
+
+        if task.name == "start":
+            self._executing_hosts_counter += 1
+
+        # NOTE assumes 2 forks
+        num_forks = 2
+        if self._executing_hosts_counter > num_forks:
+            # Exception is caught and turned into just a warning in TQM,
+            # so raise BaseException to fail the test
+            raise BaseException(
```

Comment:
Just to make sure this is a little less likely to be affected by changes elsewhere, could we also add a `display` here too?  And ensure we don't see the output. Just concerned that if something happens to how we catch exceptions, this could be swallowed, and then we'd have the potential to see false positives.